### PR TITLE
Remove mailbox API from effection-express

### PR DESCRIPTION
**Pull request by jnicklas**
Get rid of further usages of `Mailbox`.

### CODE REVIEW

**Review by cowboyd — APPROVED** [review]
Adios 👋

**Comment by jnicklas** [comment]
Made some more substantial changes to this, so would be good with a re-review. The connection server now uses the agent handler internally, so we're no longer duplicating this code.

File: packages/server/src/connection-server.ts [hunk 1]
```diff
@@ -1,71 +1,46 @@
...
+      agent.set({ ...connection.data, agentId: connection.agentId });
 
-      yield fork(function*() {
+      yield spawn(function*(): Operation<void> {
         while (true) {
-          console.debug('[connection] waiting for message', agentId);
-          let message = yield options.inbox.receive({ agentId: agentId });
-
-          yield socket.send(message);
+          let message = yield options.inbox.receive({ agentId: connection.agentId });
```
Comment (jnicklas) [inline packages/server/src/connection-server.ts:32]: We could take this even further here and make the inbox a Channel, but I wanted to leave this to a future PR.

**Review by cowboyd — APPROVED** [review]
The progress is real ❤️

**Pull request merged**